### PR TITLE
Use unique post slugs and attach to the correct organisation

### PR DESCRIPTION
**Pull request by symroe**
Refs #929

### CODE REVIEW

File: ynr/apps/elections/uk/every_election.py [hunk 1]
```diff
@@ -141,7 +136,9 @@ def get_or_create_post(self):
                 label = self["organisation"]["official_name"]
                 role = self["elected_role"]
             try:
-                self.post_object = Post.objects.get(slug=slug)
+                self.post_object = Post.objects.get(
+                    slug=slug, organization=self.organization_object
+                )
```
Comment (symroe) [inline ynr/apps/elections/uk/every_election.py]: In this I'm:

1. assuming `self["division"]["official_identifier"]` is always unique
2. also using the organisation in the look up. 

I expect I don't need to do both of these things, but making sure we have the correct organisation feels like the right thing to do here anyway. 

There is a unique constraint on `Post.slug`, but not `unique_together('slug', 'organization'). I should move to that?

**Comment by symroe** [comment]
Right, after some more thinking I realised that changing the slug _values_ at the moment isn't something I want to do. This is because they are used in various places, including CSVs and more importantly the versions JSON (Refs #697). 

Pulling back a little:

1. Things are a bit broken at the moment, and they need fixing
2. The only real problem at the moment is that posts are getting assigned to the wrong organisation in some cases
3. To fix _that and only that_ problem, we can just assign them properly without changing anything else, and deal with the rest later.

I reckon we should move to some sort of start/end date on `Post` (or maybe even copy the EE data model and use DivisionSets), and make the uniqueness based on a date range, but at the moment, this solves the problem and it's been useful to think about it a bit more.

**Comment by symroe** [comment]
This pull request **fixes 1 alert** when merging 70988dd9039d0488d219668852261200fb1810b9 into 20a92b3891c711dfd0e5aa2feebf1d871ea1f9ce - [view on LGTM.com](https://lgtm.com/projects/g/DemocracyClub/yournextrepresentative/rev/pr-85d7d28b9046aa18afd331f993f26a043e401c43)

**fixed alerts:**

* 1 for Conflicting attributes in base classes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by chris48s** [comment]
I'm going to pop out for a bit while there's still some sun :sun_behind_small_cloud:  but I'll aim to give this one a read over this evening.

Is your [test failure](https://travis-ci.org/DemocracyClub/yournextrepresentative/jobs/522979660) a fun non-deterministic ordering bug?

**Comment by chris48s** [comment]
This seems like it is sensible once you sort out the failing test.

> Right, after some more thinking I realised that changing the slug values at the moment isn't something I want to do

Totally agreed. Right now today (with 2 separate polling days coming up in the next month), we definitely just want to do the bare minimum thing that gets this fixed and allows us to clean up the mess.

Just as another thing to chuck into your thinking on this, in Every Election, when we assign a division a temp slug, it is actually a _temp_ slug in that we replace it with a GSS code as soon as one becomes available. We've never done that in YNR [but I would like to](https://github.com/DemocracyClub/yournextrepresentative/issues/553). As we think about how to do this better, it is also worth thinking about these "identifiers" as being something that can and will change later.

**Comment by symroe** [comment]
This pull request **fixes 1 alert** when merging 847ad756950a73489a6436131d1459dcfa7b0484 into 20a92b3891c711dfd0e5aa2feebf1d871ea1f9ce - [view on LGTM.com](https://lgtm.com/projects/g/DemocracyClub/yournextrepresentative/rev/pr-cd9ebb3f487cb74eff78a696f0fd384171b42443)

**fixed alerts:**

* 1 for Conflicting attributes in base classes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**